synctl

```diff
@@ -3995,19 +4152,36 @@ def main():
                 if get_args.custom_property is not None:
                     split_string = get_args.custom_property.split(',')
                     update_instance.update_custom_properties(split_string)
-                updated_payload = update_instance.get_json()
+                updated_payload = update_instance.get_updated_test_config()
                 update_instance.update_a_synthetic_test(get_args.id, updated_payload)
         if get_args.syn_type == SYN_ALERT:
-            invalid_options = [get_args.active, get_args.frequency, get_args.timeout, get_args.retry_interval, get_args.retries,
-                               get_args.operation, get_args.script_file, get_args.location, get_args.record_video,
-                               get_args.mark_synthetic_call, get_args.entry_file, get_args.url, get_args.follow_redirect,
-                               get_args.expect_status, get_args.validation_string, get_args.bundle, get_args.custom_property]
-            for option in invalid_options:
-                if option is not None:
-                    print("not supported by smart alert")
+            invalid_options = ["label", "active", "frequency", "timeout", "retry_interval", "retries", "operation", "script_file",
+                               "location", "record_video", "mark_synthetic_call", "entry_file", "url", "follow_redirect",
+                               "expect_status", "validation_string", "bundle", "custom_property"]
+            update_instance.invalid_update_options(invalid_options, update_args)
+            payload = alert_instance.retrieve_a_smart_alert(get_args.id)
+            update_alert.set_updated_payload(payload)
             if get_args.file is not None:
-                new_payload = update_instance.update_using_file(get_args.file)
-                update_instance.update_a_smart_alert(get_args.file, new_payload)
+                new_payload = update_alert.update_using_file(get_args.file)
+                update_alert.update_a_smart_alert(get_args.id, new_payload)
+            elif get_args.enable is True or get_args.disable is True:
```

Comment:
Add a filter here, since `--enable/--disable` cannot be used with these options `['name', 'test', 'alert-channel', 'severity', 'violation-count']`